### PR TITLE
Fixes edge case for streamed responses via the content app

### LINKED ISSUE #2119: DNF (still) gets wrong RPM download size for on-demand repo
Re-opening of https://pulp.plan.io/issues/9213, running the supposedly fixed pulp-rpm versions, dnf still gets the wrong sizes for RPM files downloaded from an on-demand repository. Fetching with curl or wget _sometimes_ causes dnf to then pull the correct content, but not other times. Pulp is in use here via Foreman/Katello nightly (updated regularly).

Issue reliably reproduceable with https://rpm.releases.hashicorp.com/. Every newly released RPM has the same issue.

Installed versions (fix for #9213 supposedly in 3.16.0):
- rubygem-pulp_rpm_client-3.16.1-1.el8.noarch
- python38-pulp-rpm-3.16.1-1.el8.noarch
```
[root@alma-canary ~]# dnf upgrade consul -y
Dependencies resolved.
=================================================================================================================================
 Package                 Architecture            Version                      Repository                                    Size
=================================================================================================================================
Upgrading:
 consul                  x86_64                  1.11.2-1                     sihnon_hashicorp_stable-el8                   37 M

Transaction Summary
=================================================================================================================================
Upgrade  1 Package

Total download size: 37 M
Downloading Packages:
[MIRROR] consul-1.11.2.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 38148438 but expected size is: 38490036
[MIRROR] consul-1.11.2.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 38148438 but expected size is: 38490036
[MIRROR] consul-1.11.2.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 38148438 but expected size is: 38490036
[MIRROR] consul-1.11.2.x86_64.rpm: Interrupted by header callback: Server reports Content-Length: 38148438 but expected size is: 38490036
[FAILED] consul-1.11.2.x86_64.rpm: No more mirrors to try - All mirrors were already tried without success                      

The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Error downloading packages:
  Cannot download Packages/c/consul-1.11.2.x86_64.rpm: All mirrors were tried
```

```
[root@alma-canary ~]# wget https://whitefall.jellybean.sihnon.net/pulp/content/sihnon/Library/custom/hashicorp/stable-el8/Packages/c/consul-1.11.2.x86_64.rpm
--2022-01-19 20:41:21--  https://whitefall.jellybean.sihnon.net/pulp/content/sihnon/Library/custom/hashicorp/stable-el8/Packages/c/consul-1.11.2.x86_64.rpm
Connecting to whitefall.jellybean.sihnon.net (whitefall.jellybean.sihnon.net)|81.187.154.141|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 38148438 (36M) [binary/octet-stream]
Saving to: ‘consul-1.11.2.x86_64.rpm.5’

consul-1.11.2.x86_64.rpm.5       100%[=======================================================>]  36.38M  11.5MB/s    in 3.2s    

2022-01-19 20:41:24 (11.5 MB/s) - ‘consul-1.11.2.x86_64.rpm.5’ saved [38148438/38148438]

[root@alma-canary ~]# rpm -K --nosignature consul-1.11.2.x86_64.rpm.5 
consul-1.11.2.x86_64.rpm.5: DIGESTS NOT OK
```

Meanwhile downloading directly from upstream fetches the correct size:

```
[root@alma-canary ~]# dnf install --downloadonly consul-1.11.2 --disablerepo=* --enablerepo=hashicorp
Updating Subscription Management repositories.
Hashicorp Stable - x86_64                                                                        1.0 MB/s | 671 kB     00:00    
Last metadata expiration check: 0:00:01 ago on Wed 19 Jan 2022 21:31:04 UTC.
Dependencies resolved.
=================================================================================================================================
 Package                      Architecture                 Version                         Repository                       Size
=================================================================================================================================
Upgrading:
 consul                       x86_64                       1.11.2-1                        hashicorp                        37 M

Transaction Summary
=================================================================================================================================
Upgrade  1 Package

Total download size: 37 M
DNF will only download packages for the transaction.
Is this ok [y/N]: y
Downloading Packages:
consul-1.11.2.x86_64.rpm                                                                         8.9 MB/s |  37 MB     00:04    
---------------------------------------------------------------------------------------------------------------------------------
Total                                                                                            8.9 MB/s |  37 MB     00:04     
Complete!
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.

[root@alma-canary ~]# ls -l /var/cache/dnf/hashicorp-164999f2fbadbd87/packages/
total 105192
-rw-r--r-- 1 root root 38490036 Jan 19 21:31 consul-1.11.2.x86_64.rpm
```

### PR DESCRIPTION
**Pull request by dralley**
Fixes streamed responses when the remote server declares a
content-encoding.

fixes #2119

### CODE REVIEW

File: pulpcore/content/handler.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ class Handler:
     hop_by_hop_headers = [
         "connection",
         "content-encoding",
+        "content-length",
```
Comment (dralley) [inline pulpcore/content/handler.py:98]: Not strictly true by the HTTP spec definition of "hop by hop header", but it's true in our case due to not acting as a 100% transparent proxy.  We are apparently not the first, either.

https://github.com/ledgetech/lua-resty-http/issues/155

**Review by mdellweg — COMMENTED** [review]
If i read this correctly, the difference is that the `Content-Lenght` header will be removed from the response? Don't we want to keep it if we don't have a range?

**Review by dkliban — APPROVED** [review]

**Comment by dkliban** [comment]
> If i read this correctly, the difference is that the `Content-Lenght` header will be removed from the response? Don't we want to keep it if we don't have a range?

The content-length header is going to removed from the response only when on_demand content is retrieved for the first time.

**Review by mdellweg — APPROVED** [review]
I still think we are removing the header in a case where it would not be necessary, but i am also convinced that it is always safe to remove it in the streaming response.

**Pull request merged**